### PR TITLE
Docker image

### PR DESCRIPTION
**Pull request by philpep**

### CODE REVIEW

File: ckan/install.sls [hunk 1]
```diff
@@ -35,28 +35,25 @@ ckan-user:
 ckan-venv:
   pkg.installed:
     - pkgs:
-      - python-virtualenv
       - {{ ckan.python_pip }}
 
+  cmd.run:
+    - name: pip install --user virtualenv
+    - user: {{ ckan.ckan_user }}
+    - creates: {{ home }}/.local/bin/virtualenv
```
Comment (dlax) [inline ckan/install.sls]: use `pip.installed` and pass `upgrade: true`

**Review by dlax — CHANGES_REQUESTED** [review]
As a follow-up to Solr fixes, you can look at https://hub.docker.com/r/ckan/solr/~/dockerfile/

File: ckan/install.sls (hunk 1 above)
Comment (philpep) [reply ckan/install.sls]: Unfortunately, seems that salt pip.installed has no equivalent for `--user`.

https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html#module-salt.states.pip_state

**Comment by dlax** [comment]
Philippe Pepiot a écrit :
> *@philpep* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In ckan/install.sls 
> <https://github.com/logilab/ckan-formula/pull/11#discussion_r148781489>:
> 
>>        - {{ ckan.python_pip }}
>   
> +  cmd.run:
> +    - name: pip install --user virtualenv
> +    - user: {{ ckan.ckan_user }}
> +    - creates: {{ home }}/.local/bin/virtualenv
> 
> Unfortunately, seems that salt pip.installed has no equivalent for |--user|.
> 
> https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html#module-salt.states.pip_state
> 

You can use:

   - install_options:
     - --user

That's already used in ckan/supervisor.sls.

**Comment by philpep** [comment]
On 03/11/2017, Denis Laxalde wrote:
> Philippe Pepiot a écrit :
> > *@philpep* commented on this pull request.
> > 
> > ------------------------------------------------------------------------
> > 
> > In ckan/install.sls 
> > <https://github.com/logilab/ckan-formula/pull/11#discussion_r148781489>:
> > 
> >>        - {{ ckan.python_pip }}
> >   
> > +  cmd.run:
> > +    - name: pip install --user virtualenv
> > +    - user: {{ ckan.ckan_user }}
> > +    - creates: {{ home }}/.local/bin/virtualenv
> > 
> > Unfortunately, seems that salt pip.installed has no equivalent for |--user|.
> > 
> > https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html#module-salt.states.pip_state
> > 
> 
> You can use:
> 
>    - install_options:
>      - --user
> 
> That's already used in ckan/supervisor.sls.

Didn't known that --user was a setup.py install option. Great, thanks!

**Comment by philpep** [comment]
updated / rebased the PR and added some usage of the docker image in the README

**Review by dlax — APPROVED** [review]

**Comment by philpep** [comment]
Updated to fix travis tests that run `./run-tests.py dev --cmd "salt-call ..."` where salt-call was called as ckan user (USER in Dockerfile) instead of root. Fixed by running command passed in `--cmd` as root user.

**Pull request merged**